src/lib/pages/proposal-details/components/vote-details/voting-period/validator-votes-table/index.tsx

```diff
@@ -0,0 +1,270 @@
+  isSearching: boolean;
+  isProposalResolved: boolean;
+}
+
+export const ValidatorVotesTableBody = ({
+  validatorVotes,
+  fullVersion,
+  isLoading,
+  isSearching,
+  isProposalResolved,
+  // eslint-disable-next-line sonarjs/cognitive-complexity
+}: ValidatorVotesTableBodyProps) => {
+  const isMobile = useMobile();
+
+  const templateColumns = {
+    head:
+      fullVersion && !isMobile
+        ? `${!isProposalResolved ? "0.2fr " : ""}1fr 0.8fr 1.5fr 1fr`
+        : `${!isProposalResolved ? "0.1fr " : ""}1.5fr 1fr`,
+    row:
+      fullVersion && !isMobile
+        ? `${!isProposalResolved ? "0.2fr " : ""}1fr 0.8fr 1.5fr 1fr`
+        : `${!isProposalResolved ? "0.1fr " : ""}1.5fr 1fr`,
+  };
+
+  if (isLoading) return <Loading />;
+  if (!validatorVotes) return <ErrorFetching dataName="votes" />;
+
+  if (validatorVotes.length === 0) {
+    return (
+      <EmptyState
+        imageVariant="empty"
+        message={
+          isSearching
+            ? "There are no vote matches your result."
+            : "The proposal has no votes yet."
+        }
+      />
+    );
+  }
+  return (
+    <TableContainer>
+      <ValidatorVotesTableHeader
+        templateColumns={templateColumns.head}
+        fullVersion={fullVersion}
+        isProposalResolved={isProposalResolved}
+      />
+      {validatorVotes.map((each, idx) => (
+        <ValidatorVotesTableRow
+          key={
+            each.proposalId.toString() +
+            (each.timestamp ?? "null") +
+            (each.txHash ?? "null") +
+            idx.toString()
+          }
+          proposalVote={each}
+          fullVersion={fullVersion}
+          templateColumns={templateColumns.row}
+          isProposalResolved={isProposalResolved}
+        />
+      ))}
+    </TableContainer>
+  );
+};
+
+interface ValidatorVotesTableProps {
+  id: number;
+  answers: Option<ProposalAnswerCountsResponse["validator"]>;
+  fullVersion: boolean;
+  isProposalResolved: boolean;
+  onViewMore?: () => void;
+}
+
+const tableHeaderId = "validatorVotesTable";
+
+export const ValidatorVotesTable = ({
+  id,
+  answers,
+  fullVersion,
+  isProposalResolved,
+  onViewMore,
+}: ValidatorVotesTableProps) => {
+  const [answerFilter, setAnswerFilter] = useState<ProposalValidatorVoteType>(
+    ProposalValidatorVoteType.ALL
+  );
+  const [search, setSearch] = useState("");
+  const debouncedSearch = useDebounce(search);
+
+  const {
+    pagesQuantity,
+    setTotalData,
+    currentPage,
+    setCurrentPage,
+    pageSize,
+    setPageSize,
+    offset,
+  } = usePaginator({
+    initialState: {
+      pageSize: 10,
+      currentPage: 1,
+      isDisabled: false,
+    },
+  });
+
+  const { data, isLoading } = useProposalValidatorVotes(
+    id,
+    pageSize,
+    offset,
+    answerFilter,
+    debouncedSearch
+  );
+
+  // update total data because we do filter and search on frontend side
+  useEffect(() => {
+    setTotalData(data?.total ?? 0);
+  }, [data, setTotalData]);
```

Comment:
should `useProposalValidatorVotes` take `onSuccess` and move this to inside `onSuccess`? similarly to what I did in other files e.g., `lib/pages/blocks/components/BlockTable.tsx`

Reply:
Can't use onSuccess because this query is filtering on client side.